tools/auto-derive/derive.sh

```
@@ -0,0 +1,1088 @@
+    local b
+    while IFS= read -r b; do
+        [[ -z "$b" || "$b" == "master" ]] && continue
+        local vref
+        # prefer the row whose version_ref == branch name (the "dev pointer")
+        vref="$(BRANCH="$b" yq -r '[.[] | select(.unreleased != true) | select(.branch == strenv(BRANCH)) | select(.openemr_version_ref == strenv(BRANCH))] | .[0].openemr_version_ref // ""' "$RELEASE_TARGETS_FILE")"
+        if [[ -z "$vref" || "$vref" == "null" ]]; then
+            # fall back to the first non-unreleased row's version_ref
+            vref="$(BRANCH="$b" yq -r '[.[] | select(.unreleased != true) | select(.branch == strenv(BRANCH))] | .[0].openemr_version_ref // ""' "$RELEASE_TARGETS_FILE")"
+        fi
+        [[ -n "$vref" && "$vref" != "null" ]] || fail "no version_ref resolvable for rel branch $b"
+        RT_BRANCH_VERSION_REF["$b"]="$vref"
```

Shellcheck (0.11.0)

[warning] RT_BRANCH_VERSION_REF appears unused. Verify use (or export if used externally).

(SC2034)